cql3/functions/user_function.cc

```diff
@@ -58,7 +60,40 @@ bytes_opt user_function::execute(cql_serialization_format sf, const std::vector<
         },
         [&] (wasm::context& ctx) {
             try {
-                return wasm::run_script(ctx, arg_types(), parameters, return_type(), _called_on_null_input).get0();
+                auto func_cache = ctx.cache;
+                auto [func_inst, return_promise] = func_cache->get(name(), [this, &ctx] () {
```

Comment:
Can we use the get() variant that doesn't accept a function (and uses the function from the constructor instead?) I want to get rid of the lambda variant.

Reply:
It will also be nice to extract this to a separate function regardless.

Reply:
I'll try doing that